src/components/AnchorList/index.js

```diff
@@ -0,0 +1,179 @@
+      })
+    : ''
+
+  const handelClickMobile = (e) => {
+    setOpen(true)
+    setMinisterState(links.indexOf(e.target.innerText))
+    setStateQuery(e.target.innerText)
+  }
+
+  const handelClick = (e) => {
+    setStateQuery(e.target.innerText)
+    setMinisterState(links.indexOf(e.target.innerText))
+  }
+
+  const anchorLinksMobile = links.map((link, index) => {
+    return (
+      <AnchorLinksMob key={index}>
+        <AnchorIconLink>
+          <Icon type="add" />
+        </AnchorIconLink>
+        <div className="anchor_links" onClick={handelClickMobile}>
+          {link ? link : ''}
+        </div>
+      </AnchorLinksMob>
+    )
+  })
+
+  const anchorArray = []
+
+  const anchorFirstImage = anchorQuery?.ministries[0]?.list_image?.localFile
+    ?.childImageSharp?.fluid
+    ? anchorQuery.ministries[0].list_image.localFile.childImageSharp.fluid
+    : ''
+
+  const anchorFirstText = anchorQuery?.ministries[0]?.list_text
+    ? anchorQuery.ministries[0].list_text
+    : ''
+
+  anchorQuery?.ministries
+    ? anchorQuery.ministries.map((info) => {
+        if (info.list_title.text === stateQueury) {
+          anchorArray.push(info)
+        }
+      })
+    : ''
+
+  const anchorLinks = links.map((link, index) => {
+    return (
+      <AnchorLinks key={index}>
+        <AnchorIconLink>
+          <Icon type="add-des" />
+        </AnchorIconLink>
+        <div
+          className="anchor_links"
+          style={
+            ministerState === index ? { color: '#CC1D27' } : { color: 'black' }
+          }
+          onClick={handelClick}
+        >
+          {link ? link : ''}
+        </div>
+      </AnchorLinks>
+    )
+  })
+
+  return (
+    <AncherContainer>
+      <WidthLimiterContainer>
+        <AnchorContent>
+          <Tablet>
+            <AnchorTitle as="h2" type="heading2">
+              <AnchorIconTitle>
+                {anchorQuery?.title ? anchorQuery.title : ''}
+                <Icon type="cross" />
+              </AnchorIconTitle>
+            </AnchorTitle>
+            {anchorLinksMobile ? anchorLinksMobile : ''}
+            <Modal
+              open={open}
+              setOpen={setOpen}
+              clickFunction={handelClick}
+              query={{
+                ministries: anchorQuery.ministries,
+                currentLink: links[ministerState],
+                links: links,
+                linkIndex: ministerState,
+              }}
+            />
+          </Tablet>
+          <Desktop>
+            <AnchorTitle as="h2" type="heading2">
+              <AnchorIconTitle>
+                {anchorQuery?.title ? anchorQuery.title : ''}
+                <Icon type="cross-des" />
+              </AnchorIconTitle>
+            </AnchorTitle>
+            <AnchorContainerDes>
+              <AnchorLinksContainerDes>
+                {anchorLinks ? anchorLinks : ''}
+              </AnchorLinksContainerDes>
+              <AnchorContentDes>
+                {anchorArray.length > 0 ? (
```

Comment:
length check should be anchorArray?.length > 0 to check existence of anchorArray first before checking length. Also should be outside AnchorContentDes